Tasks/IISWebAppManagementOnMachineGroup/Utility.ps1

```diff
@@ -230,41 +246,84 @@ function Repair-Inputs([ref]$siteName, [ref]$physicalPath, [ref]$poolName, [ref]
     {
         $physicalPathAuthuser.Value = $physicalPathAuthuser.Value.Trim()
     }
-    if ($sslCertThumbPrint -ne $null) 
+    if($sslCertThumbPrint -ne $null -and -not [string]::IsNullOrWhiteSpace($sslCertThumbPrint.Value))
     {
-        # Trim all non-hexadecimal characters from the ssl cetificate thumbprint
         if([regex]::IsMatch($sslCertThumbPrint.Value, "[^a-fA-F0-9]+"))
         {
-            Write-Warning (Get-VstsLocString -Key "SSLCertWarningInvalidCharacters")
+            Write-Warning (Get-VstsLocString -Key "SSLCertWarningInvalidCharactersInBinding" -ArgumentList $protocol, $ipAddress, $port)
         }
 
-        $sslCertThumbprint.Value = [Regex]::Replace($sslCertThumbprint.Value, "[^a-fA-F0-9]+" , "")
-
+        $sslCertThumbPrint.Value = [Regex]::Replace($sslCertThumbPrint.Value, "[^a-fA-F0-9]+" , "")
+        
         # Mark the SSL thumbprint value to be a secret value 
```

Comment:
This comment is not relevant for below newly introduced statements. You can move the comment closer to the below code where we set secret variable.